aa/data.py

```diff
@@ -271,4 +340,22 @@ def data_from_events(pv, events, count=None):
         timestamps[i] = event.timestamp
         severities[i] = event.severity
 
-    return ArchiveData(pv, values, timestamps, severities)
+    return ArchiveData(pv, values, timestamps, severities, enum_options)
+
+
+def parse_enum_options(meta_dict: Dict[str, str]) -> OrderedDict[int, str]:
+    """Parse enum options, if available, from a metadata dict.
+
+    Returns
+        a dict with keys of enum value (str) and values of enum string
+    """
+    output_dict = OrderedDict()
+
+    for key in sorted(meta_dict.keys()):
```

Comment:
Unfortunate, but this is a lexicographical search and puts 11 before 2.

```
>>> data.enum_options
OrderedDict([(0, 'SR'), (1, 'SRI13'), (10, 'VMXTHz'), (11, 'DIAD'), (12, 'DIADSP'), (13, 'DIADTHz'), (2, 'SRI0913'), (3, 'SRLE3ps'), (4, 'SRLEm3ps'), (5, 'SRLETHz'), (6, 'SRI0913_MOGA'), (7, 'SRI21'), (8, 'VMX'), (9, 'VMXSP')])
```

Reply:
oops